src/utils/wallet/providers/okx_wallet.ts

```diff
@@ -112,7 +112,19 @@ export class OKXWallet extends WalletProvider {
       throw new Error("OKX Wallet not connected");
     }
     // Use signPsbt since it shows the fees
-    return await this.bitcoinNetworkProvider.signPsbt(psbtHex);
+    return await this.bitcoinNetworkProvider.signPsbt(
+      psbtHex,
+      // Required on 3.29.26 version
+      {
+        toSignInputs: [
+          {
+            index: 0,
```

Comment:
This looks hacky. `signPsbt` is being used to not only sign transactions with a single input. Also, what if the input is not a taproot one?

Reply:
Agree, but it's not failing to sign the psbt
We can do a list with options in `src/app/common/utils/psbt.ts`. This `signPsbtTransaction` is used only in Unbonding and Withdrawing

So if it's OKX and we are unbonding or withdrawing, we can pass the boolean flag

Based on that flag OKX will check whether the wallet is on Taproot, and if so - add the needed options

---

Another solution is for OKX to disable the changes, but question is what is more future-proof?